ui/app/registration/views/navigationPrompt.html

```diff
@@ -0,0 +1,21 @@
+<div class="revise-refill-modal-wrapper" aria-labelledby="dialog-label">
+    <!--Added to override the z-index property of the top navigation bar. It is done to prevent any user action
+    when the popup is shown-->
+    <style type="text/css">
+        .opd-header-wrapper {
+            z-index: 10000 !important;
+        }
+    </style>
+    <div class="drug-details">
+        <p>{{'SAVE_CONFIRMATION_DIALOG_MESSAGE_KEY' | translate}}</p>
+    </div>
+
+    <div class="dialog-button-group">
+        <button id="modal-revise-button1" class="secondary-button" ng-click="continueWithoutSaving()">{{'SAVE_CONFIRMATION_OPTION_DONT_SAVE_KEY' | translate}}</button>
+        <button id="modal-revise-button2" class="secondary-button" ng-click="saveAndContinue()">{{'SAVE_CONFIRMATION_OPTION_CANCEL_KEY' | translate}}</button>
+    </div>
+
+    <span class="ngdialog-end"></span>
+</div>
```

Comment:
Was this necessary? Was there not a reusable popup directive anywhere?
Could the popup used in Clinical not be reused for example?

Reply:
I'll reuse the Clinical one.The clinical one has 3 buttons."Dont Save","Save","Cancel"
I'll write an ng-show for "Save" button to check if we are in Clinical module.If we are in Clinical Module show all 3 buttons else in Registration Module just show 2 buttons